core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestServer.scala

```diff
@@ -238,9 +238,16 @@ private[rest] class StandaloneSubmitRequestServlet(
     val sparkJavaOpts = Utils.sparkJavaOpts(conf)
     val javaModuleOptions = JavaModuleOptions.defaultModuleOptionArray().toImmutableArraySeq
     val javaOpts = javaModuleOptions ++ sparkJavaOpts ++ defaultJavaOpts ++ extraJavaOpts
+    val sparkSubmitOpts = if (mainClass.equals(classOf[SparkSubmit].getName)) {
```

Comment:
There is no problem when we call `mainClass` directly because we are in the same JVM. However, SparkSubmit is a new process which is also invoking `PythonRunner` separately. So, we need to handle the case where `SparkSubmit` is the mainClass.